Makefile

```diff
@@ -27,6 +27,14 @@ ifndef MKDOCS
 MKDOCS:=mkdocs
 endif
 
+# build output related directories
+ifdef OUT
+OUT_DIR=$(OUT)
+else
+OUT_DIR=$(CURDIR)/OUT

```

Comment:
Rather than `OUT` could we use `Outputs` or `Results`, we no longer live in 8.3 filenames and can use nicer names :)

Reply:
Good one! Just decided to "inherit" this from android/AOSP. Ok, how about `BUILDS` in caps? Just would like to make it noticeable (as something foreign to sources/scripts/configs in the tree).